BDArmory/Weapons/ModuleWeapon.cs

```diff
@@ -2080,9 +2074,9 @@ private void Fire()
 
                                     pBullet.timeElapsedSinceCurrentSpeedWasAdjusted = iTime;
                                     // measure bullet lifetime in time rather than in distance, because distances get very relative in orbit
-                                    pBullet.timeToLiveUntil = Mathf.Max(maxTargetingRange, maxEffectiveDistance) / bulletVelocity * 1.1f + Time.time;
+                                    pBullet.timeToLiveUntil = Mathf.Max(maxTargetingRange, maxEffectiveDistance) / bulletVelocity * 1.1f + Time.fixedTime;
 
-                                    timeFired = Time.time - iTime;
+                                    timeFired = Time.fixedTime + Time.fixedDeltaTime - iTime;
```

Comment:
I was not aware that `Time.time` equivalent to `Time.fixedTime` inside FixedUpdate, I can revert these changes.

However I'm somewhat less confident about the timeFired issue. Components instantiated during a FixedUpdate do not run their own FixedUpdate until the next fixed frame. To verify this, this patch logs the difference between the first fixed update time and time created for bullets:
[Log_first_bullet_fixed_update_time.patch](https://github.com/BrettRyland/BDArmory/files/12363807/Log_first_bullet_fixed_update_time.patch)
There is always a fixedDeltaTime long delay. 

`[BDArmory.PooledBullet] Time between creation and first fixed update: 0.01989746`


My conceptual understanding was that the initial bullet placement represents the bullet being fired in the upcoming physics frame, and compensating for the fact that the bullet will not run its FixedUpdate function before it. Than, every following execution of the bullets FixedUpdate function would represent the bullet simulating its movement during the upcoming physics frame.

If the initial bullet placement is supposed to represent the bullet being fired in the preceding physics frame, that implies that each PooledBullet FixedUpdate function represents the bullet "catching up" to the movement it should make on the preceding physics frame. Otherwise there is "missing frame", where the bullet does not move at all during the physics frame immediately following its creation.

As far as I understand, physics for RigidBody objects happen directly after the FixedUpdate step, I feel like delaying the physics simulation of the bullet until the next FixedUpdate step would be problematic, with potentially multiple Update calls being called in-between, but I could be wrong about this.

What I am confident about however that this is not consistent with the way the PooledBullet FixedUpdate function is structured, specifically the fact that the CheckBulletCollisions check is run **before** the MoveBullet function.

https://github.com/BrettRyland/BDArmory/blob/bfc435d488e66f68b7cb2bf0a792992f590375f5/BDArmory/Ammo/PooledBullet.cs#L422-L424

This would mean bullet collision checks happen between the current physics state of the target, and the bullet physics state from the previous physics frame, which is obviously incorrect.

Reply:
Sorry, yes, you're absolutely correct that the `timeFired` is referring to the upcoming frame and so it should have the `Time.fixedDeltaTime` added to it. I'd forgotten about the bullets not being updated on the frame they're enabled.